core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2483,6 +2484,11 @@ public RelBuilder sortLimit(int offset, int fetch, RexNode... nodes) {
    */
   public RelBuilder sortLimit(int offset, int fetch,
       Iterable<? extends RexNode> nodes) {
+    if (SortRemoveRule.shouldRemoveSortBasedOnRowCount(
+        peek().getCluster().getMetadataQuery(), peek(), fetch, offset)) {
```

Comment:
I would recommend to consider changing the order of offset/fetch to be in sync with this methods order

Reply:
+1

Reply:
Make sense, i will update the code.